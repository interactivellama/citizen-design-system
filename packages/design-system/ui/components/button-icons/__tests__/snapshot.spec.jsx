/* eslint-env jest */
import React from 'react';
import ButtonIcon from '../';
import createHelpers from '../../../../jest.helpers';
import { mount } from 'enzyme';

const { matchesMarkup } = createHelpers(__dirname);

// Bare
describe('bare button icon', () => {
  it('renders a button icon', () => matchesMarkup(<ButtonIcon />));

  xit('renders a disabled button icon', () => matchesMarkup(<ButtonIcon />));

  xit('renders hint on hover', () => {
    const wrapper = mount(
      <div className="slds-hint-parent">
        <ButtonIcon iconClassName="slds-button__icon_hint" />
      </div>
    );
    return matchesMarkup(
      wrapper
        .find('.slds-hint-parent')
        .at(0)
        .simulate('mouseEnter')
        .html()
    );
  });
});

// Container
describe('button icon with container', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon size="medium" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon size="medium" disabled />));
});

// Bordered Container
describe('button icon with bordered transparent container', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon theme="transparent" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon theme="transparent" disabled />));
});

// Bordered Container Filled
describe('button icon with bordered filled container', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon theme="neutral" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon theme="neutral" disabled />));
});

// Brand
describe('brand button icon', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon theme="brand" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon theme="brand" disabled />));
});

// Inverse
describe('button icon on inversed background', () => {
  it('renders a button icon', () =>
    matchesMarkup(<ButtonIcon theme="inverse" />));

  it('renders a disabled button icon', () =>
    matchesMarkup(<ButtonIcon theme="inverse" disabled />));
});

// Dropdown
describe('button menu icon for dropdowns', () => {
  it('renders a button menu icon', () =>
    matchesMarkup(<ButtonIcon hasDropdown theme="neutral" />));

  it('renders a button menu icon on inversed background', () =>
    matchesMarkup(
      <ButtonIcon
        hasDropdown
        theme="transparent"
        className="slds-button_icon-inverse"
      />
    ));
});
