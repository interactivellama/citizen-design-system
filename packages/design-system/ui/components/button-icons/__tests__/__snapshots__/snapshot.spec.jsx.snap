// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bare button icon renders a button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`bare button icon renders a disabled button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`bare button icon renders hint on hover 1`] = `"<div class=\\"slds-hint-parent\\"><button class=\\"slds-button slds-button_icon\\"><svg class=\\"slds-button__icon slds-button__icon_hint\\" aria-hidden=\\"true\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#settings\\"></use></svg><span class=\\"slds-assistive-text\\">Provide description of action</span></button></div>"`;

exports[`brand button icon renders a button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-brand"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`brand button icon renders a disabled button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-brand"
  disabled={true}
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon on inversed background renders a button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-inverse"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon on inversed background renders a disabled button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-inverse"
  disabled={true}
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with bordered filled container renders a button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-border-filled"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with bordered filled container renders a disabled button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-border-filled"
  disabled={true}
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with bordered transparent container renders a button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-border"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with bordered transparent container renders a disabled button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-border"
  disabled={true}
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with container renders a button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-container"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button icon with container renders a disabled button icon 1`] = `
<button
  aria-haspopup={undefined}
  className="slds-button slds-button_icon slds-button_icon-container"
  disabled={true}
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button menu icon for dropdowns renders a button menu icon 1`] = `
<button
  aria-haspopup="true"
  className="slds-button slds-button_icon slds-button_icon-more"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="slds-button__icon slds-button__icon_x-small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;

exports[`button menu icon for dropdowns renders a button menu icon on inversed background 1`] = `
<button
  aria-haspopup="true"
  className="slds-button slds-button_icon slds-button_icon-container-more slds-button_icon-inverse"
  id={undefined}
>
  <svg
    aria-hidden={true}
    className="slds-button__icon"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#settings"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="slds-button__icon slds-button__icon_x-small"
  >
    <use
      xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Provide description of action
  </span>
</button>
`;
